lib/remoteproc/remoteproc_virtio.c

```diff
@@ -179,6 +192,7 @@ static const struct virtio_dispatch remoteproc_virtio_dispatch_funcs = {
 	.get_features = rproc_virtio_get_features,
 	.read_config = rproc_virtio_read_config,
 	.notify = rproc_virtio_virtqueue_notify,
+	.notify_wait = rproc_virtio_notify_wait,
```

Comment:
As mentioned the feature is optional. seems better to declare it in `remoteproc_virtio` (similar to rpvdev_notify_func) and add a new API to set it.

Reply:
It is a element of `remoteproc_ops`, if user don't want this feature, then don't set the `remoteproc_ops->wait_notified`

Reply:
> It is a element of `remoteproc_ops`, if user don't want this feature, then don't set the `remoteproc_ops->wait_notified`

That's true only if the remoteproc_ops is used. Here is an exemple of the use of rpmsg without remoteproc_ops:
https://github.com/zephyrproject-rtos/zephyr/blob/main/samples/subsys/ipc/openamp_rsc_table/src/main_remote.c#L228

Reply:
@arnopo Hi, I want to continue this PR. You mean add new api to set the `rpvdev->wait_notifed` is add a new function called `rproc_virtio_set_wait_notifed()` or add the new argument in` rproc_virtio_create_vdev()` like this:
```
struct virtio_device *
rproc_virtio_create_vdev(unsigned int role, unsigned int notifyid,
			 void *rsc, struct metal_io_region *rsc_io,
			 void *priv,
			 rpvdev_notify_func notify,
                         rpvdev_wait_notify_func wait_notifed,
			 virtio_dev_reset_cb rst_cb)
```
Both are good for me, but I think we should set the notify and wait_notifed function in same way.

Reply:
Hi @CV-Bowen ,

I will need probably few time to remind all discussions on this PR, but concerning our point:

We are trying  to not modify the existing API. As the wait_notifed is optional, I prefer the `rproc_virtio_set_wait_notifed()` API. 

Please notice also that we added more Doxygen documentation, so this PR needs rebase and some refresh .